fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/IndexingBase.java

```diff
@@ -122,6 +123,14 @@ public abstract class IndexingBase {
         this.throttle = new IndexingThrottle(common, expectedIndexState);
         this.startingTimeMillis = System.currentTimeMillis();
         this.lastTypeStampCheckMillis = startingTimeMillis;
+        for (Index index: common.getTargetIndexes()) {
+            if (index.getType().contains("lucene")) { // Is there a better way to identify Lucene indexes without generating index maintainers?
```

Comment:
We cannot assume that online indexing will be called before the merge does. I'll keep it like this for now and maybe create a lazy evaluator later on.

Reply:
The caller must specify the indexes for any `OnlineIndexer operation`. I wouldn't like to make an exception here. 
Since `mergeIndexes` may be called externally by the API, and since it is a single transaction operation, I wouldn't like to pay the penalty of another read transaction to get a store and index maintainers while creating the `OnlineIndexer` object. 
As first version, identifying Lucene by the index type may be good enough. I just hope it would have been an enum instead of a string.